passer.py

```diff
@@ -1,59 +1,62 @@
-#!/usr/bin/python
-"""Passer learns, by watching network traffic, about the servers and clients on your network."""
-#Copyright 2008-2018, William Stearns <william.l.stearns@gmail.com>
-#Passer is a PASsive SERvice sniffer.
-#Home site http://www.stearns.org/passer/
-#Github repository https://github.com/organizations/activecm/passer/
-#Dedicated to Mae Anne Laroche.
-
-#Released under the GPL version 3:
-#This program is free software: you can redistribute it and/or modify
-#it under the terms of the GNU General Public License as published by
-#the Free Software Foundation, either version 3 of the License, or
-#(at your option) any later version.
-#This program is distributed in the hope that it will be useful,
-#but WITHOUT ANY WARRANTY; without even the implied warranty of
-#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#GNU General Public License for more details.
-#You should have received a copy of the GNU General Public License
-#along with this program.  If not, see <https://www.gnu.org/licenses/>.
-
-
-#======== Imports ========
+#!/usr/bin/env python
+"""Multiprocessing version of passer."""
+#Copyright 2018 William Stearns <william.l.stearns@gmail.com>
+#Released under GPL v3
+
+#FIXME - on KeyboardInterrupt exception, drain input queue.
+
+from __future__ import print_function
+from ipaddress import summarize_address_range, IPv4Address, IPv6Address
+from multiprocessing import Process, current_process, Manager
+import multiprocessing
 import os
-import ipaddress
 import sys
-import re
+import csv
+import fileinput
+import time
 import json
-import binascii		#converting hex text to raw bytes
-import signal		#For catching Ctrl-C
-import string		#Needed for python 2.5.2?
-import warnings		#Needed for p0f?
-import unicodedata	#Needed for removing control characters
-import pytz
-import __main__		#Needed to access variables in __main__ from functions without implicit/explicit globals
+import socket				#For dns lookups
+import codecs				#For python2 utf-8 writing
+#from scapy.all import sniff, Scapy_Exception, PcapWriter
+from scapy.all import *			#Please make sure you have an up-to-date version of scapy, at least 2.4.0
+# Imports for passive_fingerprinting feature.
+from smudge.passive_data import passive_data
+from smudge.passive_data import pull_data
+from smudge.passive_data import tcp_sig
+from smudge.signature_matching import signature
+from smudge.signature_matching import matching
+from smudge.signature_matching import query_object
+
 
+sys.path.insert(0, '.')			#Allows us to load from the current directory (There was one claim that we need to create an empty file __init__.py , but this does not appear to be required.)
```

Comment:
Consider bundling ip2asn.py as a module as you have done with smudge. This would allow you to import the file without this work around. Modules can be run with `python -m ./path/to/module`. Alternatively, move ip2asn.py into a folder `ip2asn`, and make an empty `__init__.py` file in the folder. Then, you can import the file as `ip2asn.ip2asn` and still run the file from the cli.